daemon/src/order/maker.rs

```diff
@@ -0,0 +1,385 @@
+                                return None;
+                            }
+                        };
+
+                        match SetupMsg::try_from(msg) {
+                            Ok(msg) => Some(msg),
+                            Err(e) => {
+                                tracing::error!("Failed to convert to SetupMsg: {e:#}");
+                                None
+                            }
+                        }
+                    }))
+                    .fuse(),
+                    (oracle_pk, announcement),
+                    setup_params,
+                    build_party_params,
+                    sign,
+                    Role::Maker,
+                    position,
+                    n_payouts,
+                )
+                .await?;
+
+                if let Err(e) = executor
+                    .execute(id, |cfd| cfd.complete_contract_setup(dlc))
+                    .await
+                {
+                    tracing::error!(%id, "Failed to execute contract_setup_completed: {e:#}");
+                }
+
+                anyhow::Ok(())
+            }
+        };
+
+        let err_handler = {
+            let executor = self.executor.clone();
+            move |e| async move {
+                if let Err(e) = executor
+                    .execute(id, |cfd| Ok(cfd.fail_contract_setup(e)))
+                    .await
+                {
+                    tracing::error!(%id, "Failed to execute fail_contract_setup: {e:#}");
+                }
+            }
+        };
+
+        let address = ctx.address().expect("we are alive");
+        tokio_extras::spawn_fallible(&address, task, err_handler);
+    }
+}
+
+#[xtra_productivity]
+impl Actor {
+    async fn handle(&mut self, msg: Decision) -> Result<()> {
+        let id = msg.id();
+
+        tracing::debug!("Instructed to {msg} order {id}");
+
+        let sender = self
+            .decision_senders
+            .remove(&id)
+            .context("Can't make decision on nonexistent order {id}")?;
+
+        sender
+            .send(msg.into())
+            .map_err(|_| anyhow!("Can't deliver decision on taking order {id}"))?;
+
+        Ok(())
+    }
+}
+
+#[derive(Clone, Copy)]
+pub enum Decision {
+    Accept(OrderId),
+    Reject(OrderId),
+}
+
+impl Decision {
+    fn id(&self) -> OrderId {
+        match self {
+            Decision::Accept(id) | Decision::Reject(id) => *id,
+        }
+    }
+}
+
+impl From<Decision> for protocol::Decision {
+    fn from(decision: Decision) -> Self {
+        match decision {
+            Decision::Accept(_) => protocol::Decision::Accept,
+            Decision::Reject(_) => protocol::Decision::Reject,
+        }
+    }
+}
+
+impl fmt::Display for Decision {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let s = match self {
+            Decision::Accept(_) => "accept",
+            Decision::Reject(_) => "reject",
```

Comment:
nit: why lowercase?

Reply:
I'll change it to the same name as the enum. @luckysori if this was important please let me know :)